### ISSUE TITLE
ASL Pass state schema validation reports Parameters array as invalid

### ISSUE THREAD

**Issue by kjones** [aws/aws-toolkit-vscode#2459]
Getting `Incorrect type. Expected "object".` for the Parameters in the following:
```yaml
MyPassState:
  Type: Pass
  Next: MyMapState
  Parameters:
     - Prop1: Value1
       Prop2: Value2
     - Prop1: Value3
       Prop2: Value4
```
**Expected behavior**
No error should be produced. Parameters can be an array.

**Screenshots**
![image](https://user-images.githubusercontent.com/1952793/152260169-4a6a12a0-10ab-4be2-95d3-0d90e28c2f2d.png)

-   OS: MacOS 12.1
-   Visual Studio Code Version: 1.63.2
-   AWS Toolkit Version: 1.35.0

**Comment by jormello** [comment]
Hi Ken, 

From the [Input and Output Processing section](https://states-language.net/spec.html#filters) of the Amazon States Language spec:

```
The value of "Parameters" MUST be a Payload Template which is a JSON object, whose input is the result of applying the InputPath to the raw input. If the "Parameters" field is provided, its payload, after the extraction and embedding, becomes the effective input.
```

Is there a specific use case you are looking to use that requires the `Parameters` field to be an array?

**Comment by jormello** [comment]
The schema has been updated to match the constraints applied in the service. We will separately evaluate whether this will be formalized in the ASL specification.

**Comment by yaythomas** [comment]
@justinmk3 feel free to close this issue. This functionality was released in version 1.72 of the `amazon-states-language`.
